opengever/maintenance/scripts/fix_broken_intid_references.py

```diff
@@ -37,26 +38,35 @@
 
 class IntIdReferenceFixer(object):
 
-    def __init__(self, portal, dry_run, working_dir):
+    def __init__(self, portal, dry_run, working_dir, directory_path=None):
         self.portal = portal
         self.dry_run = dry_run
         self.working_dir = working_dir
+        self.directory_path = directory_path
         self.catalog = api.portal.get_tool('portal_catalog')
         self.intid_utility = getUtility(IIntIds)
         self.objs_to_fix_path = os.path.join(working_dir, 'objs_to_fix.json')
 
     def check(self):
         broken = []
+        query = {}
+        if self.directory_path:
+            query['path'] = {'query': self.directory_path, 'depth': -1}
+
         for brain in ProgressLogger(
                 'Checking for broken intid references',
-                self.catalog.unrestrictedSearchResults(),
+                self.catalog.unrestrictedSearchResults(**query),
                 logger=logger):
-            obj = brain.getObject()
-            ref = self.intid_utility.refs.get(self.intid_utility.getId(obj))
-            obj_path = '/'.join(obj.getPhysicalPath())
-            if obj_path != ref.path:
-                broken.append(obj_path)
-                logger.info("Found broken object: {}".format(obj_path))
+            try:
+                obj = brain.getObject()
+                ref = self.intid_utility.refs.get(self.intid_utility.getId(obj))
+                obj_path = '/'.join(obj.getPhysicalPath())
+                if obj_path != ref.path:
+                    broken.append(obj_path)
+                    logger.info("Found broken object: {}".format(obj_path))
+            except Exception as e:
+                logger.error("Error processing object {}: {}".format(brain.getPath(), e))
+                continue
```

Comment:
I think it's enough to wrap the try/catch only around line nr. 62. where we use the ìntid_utility`.